backend/test/services/resource_test.py

```diff
@@ -0,0 +1,119 @@
+from backend.models.user_model import User
+from backend.entities.resource_entity import ResourceEntity
+from ...models.enum_for_models import ProgramTypeEnum
+from backend.services.resource import ResourceService
+from backend.services.tag import TagService
+from backend.services.exceptions import ResourceNotFoundException
+from . import resource_test_data
+from . import user_test_data
+from .fixtures import resource_svc, user_svc, tag_svc
+from backend.models.resource_model import Resource
+import pytest
+
+
+def test_get_resource_by_user_volunteer(resource_svc: ResourceService): 
+    """ Test getting resources by a volunteer """
+    resources = resource_svc.get_resource_by_user(user_test_data.volunteer)
+    assert len(resources) == 2
+    assert isinstance(resources[0], Resource)
+
+def test_get_resources_admin(resource_svc: ResourceService):
+    """ Test getting resources by an admin """
+    resources = resource_svc.get_resource_by_user(user_test_data.admin)
+    assert len(resources) == len(resource_test_data.resources)
+    assert isinstance(resources[0], Resource)
+
+def test_get_resources_employee(resource_svc: ResourceService):
+    """ Test getting by an employee """
+    resources = resource_svc.get_resource_by_user(user_test_data.employee)
+    assert len(resources) == 5
+    assert isinstance(resources[0], Resource)
+
+def test_create_resource_admin(resource_svc: ResourceService):
+    """ Test creating resources as an admin """
+    resource = resource_svc.create(user_test_data.admin, resource_test_data.resource6)
+    assert resource.name == resource_test_data.resource6.name
+    assert isinstance(resource, Resource)
+
+def test_create_not_permitted(resource_svc: ResourceService):
+    """ Test creating resources without permission """
+    with pytest.raises(PermissionError):
+        resource = resource_svc.create(user_test_data.volunteer, resource_test_data.resource1)
+        pytest.fail()
+
+def test_get_by_id(resource_svc: ResourceService):
+    """ Test getting a resource by id as an admin """
+    test_resource = resource_test_data.resource1
+    resource = resource_svc.get_by_id(user_test_data.admin, test_resource.id)
+    assert resource is not None
+    assert resource.id == test_resource.id
+    assert resource.name == test_resource.name
+
+def test_get_by_id_no_access(resource_svc: ResourceService):
+    """ Test getting a resourced with an id no accessible to an employee """
+    test_resource = resource_test_data.resource2
+    with pytest.raises(ResourceNotFoundException):
+        resource = resource_svc.get_by_id(user_test_data.employee, test_resource.id)
+        pytest.fail()
+
+def test_update(resource_svc: ResourceService):
```

Comment:
Maybe to ensure the stability of this test, let's call the service to get this specific resource and then compare that instead of using the resource that returns from the update method.

Reply:
I think maybe the name of the resource I used to compare was misleading. I grabbed a resource from the test data to be used in the update service, then compared the one that was returned to the one in the test data to make sure they were the same. If that isn't what you meant then let me know

Reply:
What I mean is that after you get the resource that is returned from the update method, you should get the resource again by ID to make sure it was properly changed in the database. If you look at how the update returns the updated resource, it just returns ones with the properties changed with the new ones without an additional call to the database. Just want to make sure that the data is properly changed in the database.